src/Options.jl

```diff
@@ -28,65 +28,111 @@ using ..OperatorsModule:
 using ..MutationWeightsModule: MutationWeights, mutations
 import ..OptionsStructModule: Options
 using ..OptionsStructModule: ComplexityMapping, operator_specialization
-using ..UtilsModule: max_ops, @save_kwargs
-
-"""
-         build_constraints(una_constraints, bin_constraints,
-                           unary_operators, binary_operators)
-
-Build constraints on operator-level complexity from a user-passed dict.
-"""
-function build_constraints(
-    una_constraints, bin_constraints, unary_operators, binary_operators, nuna, nbin
-)::Tuple{Array{Int,1},Array{Tuple{Int,Int},1}}
+using ..UtilsModule: max_ops, @save_kwargs, @ignore
+
+"""Build constraints on operator-level complexity from a user-passed dict."""
+@unstable function build_constraints(;
+    una_constraints,
+    bin_constraints,
+    @nospecialize(unary_operators),
+    @nospecialize(binary_operators)
+)::Tuple{Vector{Int},Vector{Tuple{Int,Int}}}
     # Expect format ((*)=>(-1, 3)), etc.
     # TODO: Need to disable simplification if (*, -, +, /) are constrained?
     #  Or, just quit simplification is constraints violated.
 
-    is_bin_constraints_already_done = typeof(bin_constraints) <: Array{Tuple{Int,Int},1}
-    is_una_constraints_already_done = typeof(una_constraints) <: Array{Int,1}
-
-    if typeof(bin_constraints) <: Array && !is_bin_constraints_already_done
-        bin_constraints = Dict(bin_constraints)
+    is_una_constraints_already_done = una_constraints isa Vector{Int}
+    _una_constraints1 = if una_constraints isa Array && !is_una_constraints_already_done
+        Dict(una_constraints)
+    else
+        una_constraints
     end
-    if typeof(una_constraints) <: Array && !is_una_constraints_already_done
-        una_constraints = Dict(una_constraints)
+    _una_constraints2 = if _una_constraints1 === nothing
+        fill(-1, length(unary_operators))
+    elseif !is_una_constraints_already_done
+        [
+            haskey(_una_constraints1, op) ? _una_constraints1[op]::Int : -1 for
+            op in unary_operators
+        ]
```

Comment:
You'll need an `else una_constraints1` condition here, otherwise `_una_constraints2` will be `nothing` when the arguments are valid.

Same for `bin_constraints2`. It should look like:

```julia
@unstable function build_constraints(;
    una_constraints,
    bin_constraints,
    @nospecialize(unary_operators),
    @nospecialize(binary_operators)
)::Tuple{Vector{Int},Vector{Tuple{Int,Int}}}
    # Expect format ((*)=>(-1, 3)), etc.
    # TODO: Need to disable simplification if (*, -, +, /) are constrained?
    #  Or, just quit simplification is constraints violated.

    is_una_constraints_already_done = una_constraints isa Vector{Int}
    _una_constraints1 = if una_constraints isa Array && !is_una_constraints_already_done
        Dict(una_constraints)
    else
        una_constraints
    end
    _una_constraints2 = if _una_constraints1 === nothing
        fill(-1, length(unary_operators))
    elseif !is_una_constraints_already_done
        [
            haskey(_una_constraints1, op) ? _una_constraints1[op]::Int : -1 for
            op in unary_operators
        ]
    else
        una_constraints1
    end

    is_bin_constraints_already_done = bin_constraints isa Vector{Tuple{Int,Int}}
    _bin_constraints1 = if bin_constraints isa Array && !is_bin_constraints_already_done
        Dict(bin_constraints)
    else
        bin_constraints
    end
    _bin_constraints2 = if _bin_constraints1 === nothing
        fill((-1, -1), length(binary_operators))
    elseif !is_bin_constraints_already_done
        [
            if haskey(_bin_constraints1, op)
                _bin_constraints1[op]::Tuple{Int,Int}
            else
                (-1, -1)
            end for op in binary_operators
        ]
    else
        bin_constraints2
    end

    return _una_constraints2, _bin_constraints2
end
```

Reply:
Thanks!